### PR TITLE
Hide icon-loading-dark after loading the image

### PR DESCRIPTION
**Pull request by manishbisht**
Fixes: #160

Licence: MIT or AGPL
### Description

Hide icon-loading-dark after loading the image and displaying it back when the image is loading.
### Tested on
- [x] Ubuntu 14.04/Chrome
### Reviewers

@oparoz

### CODE REVIEW

**Comment by jancborchardt** [comment]
@jaller94 @oparoz can you review this please?

**Comment by jaller94** [comment]
For me this only works when viewing the image for the first time.

 1. Click on a small image to view it in the gallery. (loading-icon will disappear).
 2. Press left or right to switch to another image.
 3. Switch back to your small image. This time the loading-icon won't disappear.

Can you reproduce this behaviour?

 * Ubuntu 16.10/Chrome 54
 * Ubuntu 16.10/Firefox 49

**Comment by manishbisht** [comment]
I have updated the PR can you review it again @jaller94 @oparoz

**Comment by jaller94** [comment]
It took me a long time to understand why you placed the two commands at those positions. 
A line like `// Removes loading icon if loading the image has finished and the user didn't aborted the process` would be helpful. Is the function `next` is called on every picture (also the first picture and if you go to a previous image)? That name confused me.

**Comment by manishbisht** [comment]
@jaller94 For the first command what it does is it simply hides the loading icon after successfully loading the image. and the second command show the icon back if the image is still loading and then it is again hidden after loading the image using first command.

**Comment by jaller94** [comment]
I reviewed it and tested it on my personal Nextcloud instance. I approve of your Pull Request.

Hope you can convince the auto checks and someone in the project.

File: js/slideshow.js [hunk 1]
```diff
@@ -180,6 +181,7 @@
 		 * @param {Object} next
 		 */
 		next: function (current, next) {
+			this.container.find('.icon-loading-dark').show();
```
Comment (oliverpool) [inline js/slideshow.js]: You could move this before [line 99](https://github.com/nextcloud/gallery/pull/165/commits/b0efcfad58bb9d5c2dbc9b2f0ff14c492fceb943#diff-da2846a114690f2d9e41a05c414598a7R99).

I think it then looks more natural to call `show()` and then call `hide()` on resolution of the promise.

Comment (oparoz) [inline js/slideshow.js]: I agree,  show spinner, hide image, load image, hide spinner.

**Comment by manishbisht** [comment]
I have moved the code after the mentioned line @oparoz @oliverpool

**Comment by oliverpool** [comment]
@manishbisht it [appears](https://drone.nextcloud.com/nextcloud/gallery/41) that you need to [sign off](https://stackoverflow.com/questions/1962094/what-is-the-sign-off-feature-in-git-for) your commit

**Comment by manishbisht** [comment]
Earlier this was not required. Good to see this feature here !!
It seems like I have to make the signoff commit using command git commit --signoff ?

**Comment by oliverpool** [comment]
> It seems like I have to make the signoff commit using command git commit --signoff ?

From what I understand, you need to "redo" your commit with the `git commit --signoff` command, for your commit to be signed.

**Comment by manishbisht** [comment]
@oliverpool I have signoff my commit

**Pull request merged**